x-pack/plugins/apm/common/elasticsearch_fieldnames.ts

```diff
@@ -121,8 +121,20 @@ export const HOST_HOSTNAME = 'host.hostname'; // Do not use. Please use `HOST_NA
 export const HOST_NAME = 'host.name';
 export const HOST_OS_PLATFORM = 'host.os.platform';
 export const CONTAINER_ID = 'container.id';
+export const CONTAINER = 'container';
+export const CONTAINER_IMAGE = 'container.image.name';
+
+// Kubernetes
 export const KUBERNETES = 'kubernetes';
-export const POD_NAME = 'kubernetes.pod.name';
+export const KUBERNETES_CONTAINER_NAME = 'kubernetes.container.name';
+export const KUBERNETES_DEPLOYMENT = 'kubernetes.deployment';
+export const KUBERNETES_DEPLOYMENT_NAME = 'kubernetes.deployment.name';
+export const KUBERNETES_NAMESPACE_NAME = 'kubernetes.namespace.name';
+export const KUBERNETES_NAMESPACE = 'kubernetes.namespace';
+export const KUBERNETES_POD_NAME = 'kubernetes.pod.name';
+export const KUBERNETES_POD_UID = 'kubernetes.pod.uid';
+export const KUBERNETES_REPLICASET = 'kubernetes.replicaset';
+export const KUBERNETES_REPLICASET_NAME = 'kubernetes.replicaset.name';
 
 export const CLIENT_GEO_COUNTRY_ISO_CODE = 'client.geo.country_iso_code';
 
```

Comment:
This file should only contain APM fields. We may want to have a separate file per type. You can create a new folder with this structure:

```


common/
├─ es_fields
│  ├─ apm.ts
│  ├─ infra_metrics.ts
```

Reply:
I would rather split  them in a follow up PR as  [elasticsearch_fieldnames.ts](https://github.com/elastic/kibana/pull/139612/files/3b2de2e498509115a7450911656dc4d198c62899#diff-c5787c4c77787961ecc0a8227dad0835ffb21d2006dd12f768d3ce17b5a917d0) contains already fields from other types (rum , ecs).

Reply:
RUM was/is an APM agent - but I agree that RUM specific fields should perhaps go into a separate file. Same could be said for other agents eg. if we have java specific fields.

Not sure what you mean with ecs. Do we have ecs fields which are not in the APM mappings?

Reply:
sorry I missed the question. 
Elastic common schema (ecs). 

I thought the following: https://github.com/elastic/kibana/blob/35010c1f36340d3dd0b72e67c3759b42a4a8c94e/x-pack/plugins/apm/common/elasticsearch_fieldnames.ts#L119-L122
 were not in APM mapping but just checked it and we have it

Reply:
ticket: https://github.com/elastic/kibana/issues/140925